smartapps/initialstate-events/initial-state-event-streamer.src/initial-state-event-streamer.groovy

```diff
@@ -330,57 +330,54 @@ def genericHandler(evt) {
 // This is a handler function for flushing the event buffer
 // after a specified amount of time to reduce the load on ST servers
 def flushBuffer() {
+	def eventBuffer = atomicState.eventBuffer
 	log.trace "About to flush the buffer on schedule"
-	if (atomicState.eventBuffer != null && atomicState.eventBuffer.size() > 0) {
-		tryShipEvents()
+	if (eventBuffer != null && eventBuffer.size() > 0) {
+		atomicState.eventBuffer = []
+		tryShipEvents(eventBuffer)
 	}
 }
 
 def eventHandler(name, value) {
-	log.debug atomicState.eventBuffer
-
-	def eventBuffer = atomicState.eventBuffer
 	def epoch = now() / 1000
-	
-	// if for some reason this code block is being run
-	// but the SmartApp wasn't propery setup during install
-	// we need to set initialize the eventBuffer.
-	if (!atomicState.eventBuffer) {
-		atomicState.eventBuffer = []
-	}
+	def eventBuffer = atomicState.eventBuffer ?: []
 	eventBuffer << [key: "$name", value: "$value", epoch: "$epoch"]
-	
-	log.debug eventBuffer
-
-	atomicState.eventBuffer = eventBuffer
 
 	if (eventBuffer.size() >= 10) {
-		tryShipEvents()
+		// Clear eventBuffer right away since we've already pulled it off of atomicState to reduce the risk of missing
+		// events.  This assumes the grokerSubdomain, accessKey, and bucketKey are set correctly to avoid the eventBuffer
+		// from growing unbounded.
+		atomicState.eventBuffer = []
+		tryShipEvents(eventBuffer)
 	}
+	log.debug "Event added to buffer: " + eventBuffer
 }
 
 // a helper function for shipping the atomicState.eventBuffer to Initial State
-def tryShipEvents() {
+def tryShipEvents(eventBuffer) {
 
+	def grokerSubdomain = atomicState.grokerSubdomain
```

Comment:
Pull grokerSubdomain off of atomicState here and assign to variable here to avoid hitting the DB twice to check != null and size.